x-pack/plugins/apm/server/routes/services/get_services_detailed_statistics/get_service_aggregated_transaction_detailed_statistics.ts

```diff
@@ -0,0 +1,232 @@
+}: {
+  serviceNames: string[];
+  environment: string;
+  kuery: string;
+  setup: Setup;
+  searchAggregatedServiceMetrics: boolean;
+  offset?: string;
+  start: number;
+  end: number;
+  randomSampler: RandomSampler;
+}) {
+  const { apmEventClient } = setup;
+  const { offsetInMs, startWithOffset, endWithOffset } = getOffsetInMs({
+    start,
+    end,
+    offset,
+  });
+
+  const metrics = {
+    avg_duration: {
+      avg: {
+        field: TRANSACTION_DURATION_SUMMARY,
+      },
+    },
+    total_doc: {
+      value_count: {
+        field: TRANSACTION_DURATION_SUMMARY,
+      },
+    },
+    failure_count: {
+      sum: {
+        field: TRANSACTION_FAILURE_COUNT,
+      },
+    },
+    success_count: {
+      sum: {
+        field: TRANSACTION_SUCCESS_COUNT,
+      },
+    },
+  };
+
+  const response = await apmEventClient.search(
+    'get_service_aggregated_transaction_detail_stats',
+    {
+      apm: {
+        events: [ProcessorEvent.metric],
+      },
+      body: {
+        size: 0,
+        query: {
+          bool: {
+            filter: [
+              { terms: { [SERVICE_NAME]: serviceNames } },
+              ...[{ exists: { field: TRANSACTION_DURATION_SUMMARY } }],
+              ...rangeQuery(startWithOffset, endWithOffset),
+              ...environmentQuery(environment),
+              ...kqlQuery(kuery),
+            ],
+          },
+        },
+        aggs: {
+          sample: {
+            random_sampler: randomSampler,
+            aggs: {
+              services: {
+                terms: {
+                  field: SERVICE_NAME,
+                  size: serviceNames.length,
+                },
+                aggs: {
+                  transactionType: {
+                    terms: {
+                      field: TRANSACTION_TYPE,
+                    },
+                    aggs: {
+                      ...metrics,
+                      timeseries: {
+                        date_histogram: {
+                          field: '@timestamp',
+                          fixed_interval:
+                            getBucketSizeForAggregatedTransactions({
+                              start: startWithOffset,
+                              end: endWithOffset,
+                              numBuckets: 20,
+                              searchAggregatedServiceMetrics,
+                            }).intervalString,
+                          min_doc_count: 0,
+                          extended_bounds: {
+                            min: startWithOffset,
+                            max: endWithOffset,
+                          },
+                        },
+                        aggs: metrics,
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    }
+  );
+
+  return keyBy(
+    response.aggregations?.sample.services.buckets.map((bucket) => {
+      const topTransactionTypeBucket =
```

Comment:
If we don't use _doc_count, you have to add a value count sub agg and use a bucket sort agg to sort the transaction type buckets based on the value count sub aggs. Otherwise you'll sort on most documents, not on most transactions. If you have a type of transaction that is highly compressed it might drop in the list because there are less metric documents but they represent more transactions. There might be some other places where you'd need to handle this as well, the gist of it is that you can no longer depend on the default sort of a terms agg if we don't use `_doc_count`.

Reply:
What do you mean if we don't use `_doc_count`? in the apm-server? service metrics return buckets doc_count

Reply:
https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-doc-count-field.html

Reply:
based on the reference and the response I understand that we use it. Please correct if I misunderstood

Reply:
The response from the value count agg and doc_count should be the same. Which I think it isn't, based on the example you gave. If you look at the source for a document, does it include a `_doc_count` field?

Reply:
`service metrics` source doesn't include `_doc_count` field. so we don't use it. 

In comparison `transaction metrics` source does have it and `doc_count` and `value_count` are the same.

Reply:
👍 can you talk to Martijn/APM Server team if we can include that? I'm hoping that's an easy fix for them. We can change it on our side after FF, as far as I'm concerned.

Reply:
Sure, will reach out. 

for the time being, I'm practising my query skills https://github.com/elastic/kibana/pull/140868/commits/13611eb8a11dbb8bf4b47c52cd5893615fcfdd5e . Hopefully I get it right 🙏

Reply:
Opened Goland before lunch to try and get `_doc_count` in :)

Reply:
Opened PR: https://github.com/elastic/apm-server/pull/9143